libc-top-half/musl/src/env/__init_tls.c

```diff
@@ -31,25 +31,35 @@ extern unsigned char __global_base;
 extern weak unsigned char __stack_high;
 extern weak unsigned char __stack_low;
 
-static inline void setup_default_stack_size()
+struct stack_bounds {
+	void *base;
+	size_t size;
+};
+
+static inline struct stack_bounds get_stack_bounds()
 {
-	ptrdiff_t stack_size;
+	struct stack_bounds bounds;
 
-	if (&__stack_high)
-		stack_size = &__stack_high - &__stack_low;
-	else {
+	if (&__stack_high) {
+		bounds.base = &__stack_high;
+		bounds.size = &__stack_high - &__stack_low;
+	} else {
 		unsigned char *sp;
 		__asm__(
 			".globaltype __stack_pointer, i32\n"
 			"global.get __stack_pointer\n"
 			"local.set %0\n"
 			: "=r"(sp));
-		stack_size = sp > &__global_base ? &__heap_base - &__data_end : (ptrdiff_t)&__global_base;
+		if (sp > &__global_base) {
+			bounds.base = &__heap_base;
+			bounds.size = &__heap_base - &__data_end;
+		} else {
+			bounds.base = &__global_base;
+			bounds.size = (size_t)&__global_base;
+		}
```

Comment:
This code only calculates the stack found of the main thread (i.e. the singular stack that the linker produces).  Is that intended?

Reply:
This fills data for main thread, spawned threads have these fields filled in pthread_create.